### PR TITLE
Feature/loyalty cancel participation

### PR DESCRIPTION
**Pull request by Ranatamerr**
This PR implements backend functionality that allows vendors to cancel their participation in the GUC loyalty program. It introduces a new route, controller, and service method to handle cancellation requests securely and efficiently.

Key Changes:

Service Layer: Added a method to cancel an active loyalty program for a vendor. This method checks if the vendor has a pending or verified program, updates the status to “cancelled,” and records the cancellation timestamp.

Controller: Added a method to handle cancellation requests from authenticated vendors and return appropriate success or error responses.

Routes: Added a new protected DELETE endpoint /api/loyalty-partners/cancel accessible only to vendors.

Testing Instructions:

Use the DELETE endpoint with a valid vendor JWT token.

Verify that the loyalty program status updates to “cancelled” and the cancellation timestamp is recorded.

Test edge cases: vendor already cancelled, rejected vendors, missing or invalid JWT, and non-existent vendor records.

### CODE REVIEW

File: server/src/features/loyaltyPartners/loyaltyPartner.service.js [hunk 1]
```diff
@@ -34,4 +34,29 @@ export const LoyaltyPartnerService = {
 
     return newApplication;
   },
+
+  async cancelLoyaltyProgram(vendorId) {
+    // Find the active loyalty program for the vendor
+    const existing = await LoyaltyPartner.findOne({
+      vendorId,
+      status: { $in: ["pending", "verified"] },
```
Comment (JomanaMahmoud) [inline server/src/features/loyaltyPartners/loyaltyPartner.service.js]: Vendor can cancel only if active.
i changed the status enum in the model as its only active or cancelled as the admin doesnt accept or reject, the vendor joins directly

Comment (JomanaMahmoud) [reply server/src/features/loyaltyPartners/loyaltyPartner.service.js]: i also removed the uniquness in the vendorId as this prevents reapplication which might cause a problem here `const existing = await LoyaltyPartner.findOne({ vendorId });` ,  make sure you search by the vendor id + status= active

File: server/src/features/loyaltyPartners/loyaltyPartner.service.js [hunk 2]
```diff
@@ -34,4 +34,29 @@ export const LoyaltyPartnerService = {
...
+
+    if (!existing) {
+      const err = new Error("No active loyalty program found to cancel.");
+      err.statusCode = 404;
+      throw err;
+    }
+
+    // Update status to cancelled
+    existing.status = "cancelled";
+    existing.deletedAt = new Date();
```
Comment (JomanaMahmoud) [inline server/src/features/loyaltyPartners/loyaltyPartner.service.js]: remove this
```suggestion
    existing.deletedAt = new Date();
```

**Review by JomanaMahmoud — CHANGES_REQUESTED** [review]

**Review by JomanaMahmoud — APPROVED** [review]
🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈
